config/config-network.yaml

```diff
@@ -173,12 +173,25 @@ data:
     # Knative doesn't know about that otherwise.
     default-external-scheme: "http"
 
-    # internal-encryption indicates whether internal traffic is encrypted or not.
-    # If this is "true", the following traffic are encrypted:
-    #  - ingress to activator
-    #  - ingress to queue-proxy
-    #  - activator to queue-proxy
+    # internal-encryption applies to the knative control-plane only
```

Comment:
Wait -- it seems like you're re-purposing this flag to do something different?

It seems like this should continue to control the data plane server-authenticated encryption path it does today, but I'm 100% on a better flag going forward using the below.

I'd even be okay with something like:

> Setting this flag to "true" is equivalent to setting `internal-trust` to `minimal`.

You could also say:

> This flag acts as a global disable for the internal-trust feature below.  If this flag is set to "false", the "internal-trust" feature will be disabled.

Reply:
> Wait -- it seems like you're re-purposing this flag to do something different?
Yes, this is what I was proposing. 
Before the change, this flag affected both data-plane and control-plane and I am suggesting it will be dedicated to control-plane only. 

> 
> It seems like this should continue to control the data plane server-authenticated encryption path it does today, but I'm 100% on a better flag going forward using the below.
I am proposing that this flag be changed (we are in alpha, so we can still do that) and for data-plane the functionality will be moved to the other flag we now add

Reply:
The flag is no longer repurposed - it is deprecated and separated to two flags for DP and CP, each with multiple options - this will allow growing the functionality in a gradual non disruptive way